### DISCUSSION TITLE
RAID resync on new installation

### DISCUSSION THREAD

**Discussion by bbotts77** [1000001101000/Debian_on_Buffalo#107] [Q&A]
I'm guessing I'm doing something dumb here. I have a LS441D and 4-6TB WD Reds. They used to be in a RAID-10 array and I decided I would go RAID-5 when switching to DebianOnBuffalo. I forgot there was a migration path from RAID-10 to RAID-5 and decided to just back everything up to a 12TB drive and start fresh. I got the new boot images moved over and SSHed to the Debian installer. I deleted all of the pre-existing volumes and software RAID devices. I did some reading on what people recommend for partitioning lately and decided to go with a 1GB (4x512MB) RAID-10 ext3 volume for /boot, a 4GB (4x1GB) RAID-0 swap volume at the end of the available space, and the remaining  as a RAID-5 ext4 volume for /(root) and everything else, which ends up being about 18TB. Everything seemed to be going great until it got to the ext4 volume. I thought it had locked up. Then, after a few hours it dropped out of the installer screen and it looked like it started killing processes one of which was mkfs.ext4. It looked like everything had stopped. However, I decided to dig around a bit and found that md2 (the big RAID device) was doing a resync. It seems to be getting through about 2% per day. So, I'm about 45 days away from being able to proceed with any other troubleshooting. I don't think that's practical at all. Any suggestions?

**Comment by 1000001101000** [comment]
Good to hear from you!

There are some issues with this setup that I would recommend changing, one of them may even be the cause of the issues you are seeing.

The #1 issue is that 32-bit kernels such as this one have a maximum volume size of 16TB. It's possible the re-sync issues you are seeing  are related to it hitting against that limit though I am not certain. It may also be advisable to skip creating that volume in the installer and do it within Debian after the install. I believe there is a way to tell it not to do a full resync when you create a new volume but i don't have much experience with raid5. 

I'd personally recommend switching to RAID 1 for /boot, swap, and rootfs. That allows the device to still work with as little as 1 drive inserted which can be useful in some scenarios. The RAID0 swap removes the possibility of hot swapping drives, which may or may not be a big deal.

**Reply by bbotts77** [reply]
Thanks, man. I thought I responded to this last week. But, I don't see my response in here.

I completely forgot about the 16TB limit and that this would be a 32-bit device.

Thanks for the advice on the RAID levels for the partitions. For swap, I guess I was thinking more about performance, than it continuing to work during a drive failure. Silly me.

**Reply by 1000001101000** [reply]
I imagine it depends on the specifics of your use case. 

With raid0 swap the device would likely crash if you lost a drive but it would still be able to boot without it (you just wouldn't have swap). If your use case involves a heavy amount of swapping the performance increase might be worth the trade-off. The same applies to the rootfs, if you need the extra performance for some reason it might be worth it to trade that for crash resiliency. /boot should probably always be raid 1 since the boot loader likely can't read from all the drives in  RAID-10 and it doesn't get used much.

I should probably point out that being able to boot with missing drives will only work if your data array is mounted as "optional". The default behavior for debian is to drop to a recovery shell if any of the volumes is fstab fail to mount, since this device won't have a monitor/terminal that ends up being less than helpful.

**Comment by 1000001101000** [comment]
How did this turn out?

**Reply by bbotts77** [reply]
I ended up buying a new box, since I needed it to work. I still have the LS441D, though. When I upgrade the drives in the Synology box, I'll start working on this again with the old drives. Thanks again for your help on trying to get it setup.
